google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc

```diff
@@ -331,18 +346,48 @@ void RunThread(ThroughputOptions const& options, std::string const& bucket_name,
 
   std::uniform_int_distribution<std::int64_t> size_generator(
       options.minimum_object_size, options.maximum_object_size);
-  std::uniform_int_distribution<std::size_t> write_buffer_size_generator(
-      options.minimum_write_buffer_size / options.write_buffer_quantum,
-      options.maximum_write_buffer_size / options.write_buffer_quantum);
-  std::uniform_int_distribution<std::size_t> read_buffer_size_generator(
-      options.minimum_read_buffer_size / options.read_buffer_quantum,
-      options.maximum_read_buffer_size / options.read_buffer_quantum);
+
+  auto quantized_range_generator = [](auto minimum, auto maximum,
+                                      auto quantum) {
+    auto distribution = std::uniform_int_distribution<decltype(quantum)>(0, 0);
+    if (quantum != 0) {
+      distribution = std::uniform_int_distribution<decltype(quantum)>(
+          minimum / quantum, maximum / quantum);
+    }
+    return [d = std::move(distribution), quantum](auto& g) mutable {
+      return quantum * d(g);
+    };
+  };
+
+  auto write_buffer_size_generator = quantized_range_generator(
```

Comment:
in a future PR, you might want to group quantized options in a struct. Then have member functions for things like `output_quantized_*`, `quantized_range_generator`, parsing values from the command line.